pkg/scalers/influxdb_scaler.go

```diff
@@ -21,18 +24,58 @@ type influxDBScaler struct {
 	logger     logr.Logger
 }
 
+type influxDBScalerV3 struct {
+	client     influxdb3.Client
+	metricType v2.MetricTargetType
+	metadata   *influxDBMetadata
+	logger     logr.Logger
+}
+
 type influxDBMetadata struct {
-	AuthToken                string  `keda:"name=authToken, order=triggerMetadata;resolvedEnv;authParams"`
-	OrganizationName         string  `keda:"name=organizationName, order=triggerMetadata;resolvedEnv;authParams"`
-	Query                    string  `keda:"name=query, order=triggerMetadata"`
-	ServerURL                string  `keda:"name=serverURL, order=triggerMetadata;authParams"`
-	UnsafeSsl                bool    `keda:"name=unsafeSsl, order=triggerMetadata, optional"`
-	ThresholdValue           float64 `keda:"name=thresholdValue, order=triggerMetadata, optional"`
-	ActivationThresholdValue float64 `keda:"name=activationThresholdValue, order=triggerMetadata, optional"`
+	AuthToken                string  `keda:"name=authToken,                order=triggerMetadata;resolvedEnv;authParams"`
+	OrganizationName         string  `keda:"name=organizationName,         order=triggerMetadata;resolvedEnv;authParams, optional"`
+	Query                    string  `keda:"name=query,                    order=triggerMetadata"`
+	InfluxVersion            string  `keda:"name=influxVersion,            order=triggerMetadata,                        default=2"`
+	QueryType                string  `keda:"name=queryType,                order=triggerMetadata,                        default=influxql"`
```

Comment:
I think that we can validate the values here directly, can't @wozniakjan @SpiritZhou ?

```suggestion
	InfluxVersion            string  `keda:"name=influxVersion,            order=triggerMetadata,                        default=2, enum=2;3"`
	QueryType                string  `keda:"name=queryType,                order=triggerMetadata,                        default=influxql" enum=influxql;flightsql`
```

This is useful to know valud values on openspec generation (If I'm not wrong and we can do it via validation)

Reply:
there is still a comma missing in the last proposal

Reply:
The `Enum` struct tag validation appears to be case sensitive. I don't see a way to make it case insensitive. For the `QueryType` field this could be an issue, I've found users often preference towards the more readable form (e.g. `InfluxQL`).

In this case should I leave out the `enum` and continue to use the `Validate()` function to check this value. If you're open to it I could update the `enum` validation to be case insensitive, or add a new param to toggle case sensitivity on enums (or both), but that also feels way outside the scope of the PR and should probably be a decision discussed in a separate issue.

Reply:
Happy to open an issue and another PR to make `enum` case insensitive if there is general approval!

Reply:
> Perhaps a separate PR that lowers all enums, and then a change in TypedConfig to lower the valFromConfig value as well: `s := strings.ToLower(strings.TrimSpace(s))` That's a fairly small PR. Once it's merged, it can be retrieved again in this PR?

WDYT? @kedacore/keda-core-contributors

Reply:
Yes, an option is indeed a new parameter.
Or something with a prefix is ​​also possible, for example:

`enum=ci:influxql;ci:flightsql`

That way we indicate that it's **c**ase**i**nsensitive.
Then it can remain in the enum code itself, but with a small addition regarding `strings.HasPrefix(e, "ci:")`.

I have no preference.